src/querystring.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{error::Error, Context};
+use http::StatusCode;
+use serde::Deserialize;
```

Comment:
Could you perhaps, switch this to `serde::de::Deserialize` instead? (https://github.com/serde-rs/serde/issues/1441)

Reply:
This already matches the preferred import style of Serde in Rust 2018.

Reply:
uh, really? I'm curious to know where you found this. Perhaps you're confusing the macro from the trait? 

Everywhere in tide, @aturon seem to have used it explicitly: https://github.com/rustasync/tide/blob/master/src/forms.rs#L14

Everywhere in serde docs: it's used explicitly: 
https://serde.rs/impl-serialize.html

I'm confused where you got this notion from. (Anyway, while this was just a suggestion - but I'd appreciate it if you don't just mark things as resolved in a whim without either proper rationale or providing a chance to respond)

Reply:
[`Deserialize` is exported at the crate root.](https://github.com/serde-rs/serde/blob/master/serde/src/lib.rs#L231)

It's my understanding that when the `derive` feature is implemented, it also become a valid import for usage as `#[derive(Deserialize)]`. Tests pass, which seem to confirm this understanding.

It makes sense that Serde would use the full path in their docs, because it's technical writing and they need to be exact. But `Deserialize` is also an often-used API, which is why it's exported at the crate root: as a convenience for users. In our case, I'd suggest it pays dividends to think like the user (since that's essentially what we are in this context) and unify import paths where reasonable.